### PR TITLE
fix: fix tests

### PR DESCRIPTION
**Pull request by aleem1314**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -251,13 +251,25 @@ test-race: TEST_PACKAGES=$(PACKAGES_NOSIMULATION)
 
 $(TEST_TARGETS): run-tests
 
+SUB_MODULES = $(shell find . -type f -name 'go.mod' -printf '%h\n' | sort)
+CURRENT_DIR = $(shell pwd)
 run-tests:
 ifneq (,$(shell which tparse 2>/dev/null))
 	#go test -mod=readonly -json $(ARGS) $(TEST_PACKAGES) | tparse
-	find . -name go.mod -execdir go test ./... \;
+	# find . -name go.mod -execdir go test ./... \;
```
Comment (aleem1314) [inline Makefile:259]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -251,13 +251,25 @@ test-race: TEST_PACKAGES=$(PACKAGES_NOSIMULATION)
...
+	@echo "Unit tests"; \
+	for module in $(SUB_MODULES); do \
+		cd ${CURRENT_DIR}/$$module; \
+		go test -mod=readonly -json $(ARGS) $(TEST_PACKAGES) ./... | tparse; \
+	done
 else
 	#go test -mod=readonly $(ARGS) $(TEST_PACKAGES)
-	find . -name go.mod -execdir go test ./... \;
+	# find . -name go.mod -execdir go test ./... \;
```
Comment (aleem1314) [inline Makefile:267]:
```suggestion
```

File: x/group/types_test.go [hunk 3]
```diff
@@ -395,14 +395,14 @@ func TestGroupAccountInfo(t *testing.T) {
 			timeout:      proto.Duration{Seconds: 1},
 			expErr:       true,
 		},
-		"invalid group account address": {
+		"valid variable length group account address": {
 			group:        1,
 			groupAccount: []byte("any-invalid-group-address"),
```
Comment (blushi) [inline x/group/types_test.go:400]: maybe change the string here too? (removing "invalid" which makes it a bit confusing now)

File: x/group/types_test.go [hunk 4]
```diff
@@ -420,14 +420,14 @@ func TestGroupAccountInfo(t *testing.T) {
 			timeout:      proto.Duration{Seconds: 1},
 			expErr:       true,
 		},
-		"invalid admin account address": {
+		"valid variable length admin account address": {
 			group:        1,
 			groupAccount: []byte("valid--group-address"),
 			admin:        []byte("any-invalid-admin-address"),
```
Comment (blushi) [inline x/group/types_test.go:426]: ditto

File: x/group/msgs_test.go [hunk 5]
```diff
@@ -146,13 +146,6 @@ func TestMsgCreateGroupAccount(t *testing.T) {
 			timeout:   proto.Duration{Seconds: 1},
 			expErr:    true,
 		},
-		"valid admin required": {
-			admin:     []byte("invalid-address"),
```
Comment (amaurym) [inline x/group/msgs_test.go:150]: why is this test deleted?

**Review by amaurym — APPROVED** [review]

File: app/modules_test.go [hunk 6]
```diff
@@ -1,4 +1,5 @@
 // +build experimental
```
Comment (blushi) [inline app/modules_test.go:1]: This is more coming from https://github.com/regen-network/regen-ledger/pull/322 but this means now that the `testutil/network` tests are now only running in experimental mode, which weren't the case and doesn't really make sense actually, should we have a separate test file for this instead?

File: Makefile [hunk 7]
```diff
@@ -251,13 +251,23 @@ test-race: TEST_PACKAGES=$(PACKAGES_NOSIMULATION)
 
 $(TEST_TARGETS): run-tests
 
+SUB_MODULES = $(shell find . -type f -name 'go.mod' -printf '%h\n' | sort)
```
Comment (blushi) [inline Makefile:254]: this is causing some issues on mac osx `find: -printf: unknown primary or operator`

Comment (aleem1314) [reply Makefile]: Updated. can u test this again?

File: x/group/msgs_test.go (hunk 5 above)
Comment (aleem1314) [reply x/group/msgs_test.go:150]: Now we have a variable-length address, this test never fails.

**Review by blushi — APPROVED** [review]
LGTM, thanks @aleem1314!

**Review by cyberbono3 — APPROVED** [review]
LGTM. Thanks for doing that, @aleem1314 !

**Pull request merged**